mobile/library/objective-c/EnvoyEngineImpl.mm

```diff
@@ -521,17 +522,43 @@ - (int)runWithTemplate:(NSString *)yaml
   if (config.enablePlatformCertificateValidation) {
     register_apple_platform_cert_verifier();
   }
+}
+
+- (int)runWithBootstrapConfig:(EnvoyConfiguration *)config logLevel:(NSString *)logLevel {
+  auto bootstrap = [config generateBootstrap];
+  if (bootstrap == nullptr) {
+    return kEnvoyFailure;
+  }
 
-  return [self runWithConfigYAML:resolvedYAML logLevel:logLevel];
+  [self performRegistrationsForConfig:config];
+  [self startObservingLifecycleNotifications];
+
+  @try {
+    auto options = std::make_unique<Envoy::OptionsImpl>();
+    options->setConfigProto(std::move(bootstrap));
+    options->setLogLevel(options->parseAndValidateLogLevel(logLevel.UTF8String));
+    options->setConcurrency(1);
+    return reinterpret_cast<Envoy::Engine *>(_engineHandle)->run(std::move(options));
+  } @catch (NSException *exception) {
+    NSLog(@"[Envoy] exception caught: %@", exception);
+    [NSNotificationCenter.defaultCenter postNotificationName:@"EnvoyError" object:self];
+    return kEnvoyFailure;
+  }
 }
 
-- (int)runWithConfigYAML:(NSString *)configYAML logLevel:(NSString *)logLevel {
+- (int)runWithTemplate:(NSString *)yaml
+                config:(EnvoyConfiguration *)config
+              logLevel:(NSString *)logLevel {
+  NSString *resolvedYAML = [config resolveTemplate:yaml];
+  if (resolvedYAML == nil) {
+    return kEnvoyFailure;
+  }
+
+  [self performRegistrationsForConfig:config];
   [self startObservingLifecycleNotifications];
 
-  // Envoy exceptions will only be caught here when compiled for 64-bit arches.
```

Comment:
This comment is no longer relevant as we don't compile for any architectures other than 64-bit.